packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -1168,7 +1168,7 @@ class LocalizationsGenerator {
       .replaceAll('\n\n\n', '\n\n');
   }
 
-  bool _containsPluralMessage() => _allMessages.any((Message message) => message.isPlural);
+  bool _requiresIntlImport() => _allMessages.any((Message message) => message.isPlural || message.placeholdersRequireFormatting);
```

Comment:
is the plan to keep adding onto this or is there a comprehensive list of intl type usage?

Reply:
As far as I can tell, it's a case by case decision whether a feature needs intl or not. If we add more features, this list would grow.